### DISCUSSION TITLE
bilibili now has official macOS client while cask points to a 3rd party one

### DISCUSSION THREAD

**Discussion by ApolloZhu** [Homebrew/discussions#3287] [Casks]
The new, official client for bilibili can be found at https://app.bilibili.com/ with the actual download link of https://dl.hdslb.com/mobile/fixed/pc_electron_mac/bili_mac.dmg, but the `bilibili` cask https://formulae.brew.sh/cask/bilibili points to a 3rd party one (that hasn't been updated for a long time) instead https://github.com/typcn/bilibili-mac-client.

Since this is not really a bug or a security vulnerability, I've opened this discussion asking what's the best way to handle the current situation.

**Comment by SMillerDev** [comment]
Make a new cask for the official one.

**Reply by ApolloZhu** [reply]
Does it make sense for the official one to take the `bilibili` token and rename the 3rd party one to something like `typcn-bilibili`?

In my opinion the situation will be similar to https://formulae.brew.sh/cask/zoomus if the official one don't take the trademarked name (or I guess per potentially misleading names https://docs.brew.sh/Cask-Cookbook#potentially-misleading-name and that the original 3rd party one seems to be unmaintained)

**Reply by carlocab** [reply]
It's hard to say that they're the same, since the `zoom` vs `zoomus` case was primarily motivated by the fact that there were very many downloads for `zoomus`, and also many installs for `zoom` that actually wanted to install `zoomus`. This may well turn out to be the case for `bilibili`, but right now we don't really know that yet.

**Reply by ApolloZhu** [reply]
1. Can I argue that the existing vendor is not authorized to use the name `bilibili` since they don't own the trademark, and thus a prefix must be added for disambiguation?
2. If not, would `bilibili-bilibili` be an appropriate token for the official client, since vendor name is bilibili, app name is 哔哩哔哩.app (Chinese letters for bilibili)?

**Reply by ApolloZhu** [reply]
I guess the discussions will happen more in https://github.com/Homebrew/homebrew-cask/pull/124073 instead.
